File: src/backend/CareFirstClinic.API/CareFirstClinic.API/Models/Patient.cs
using System.ComponentModel.DataAnnotations;

namespace CareFirstClinic.API.Models
{

    public class Patient
    {
        public Guid Id { get; set; } = Guid.NewGuid();
        public string? AvatarUrl { get; set; }

        [Required]
        [MaxLength(100)]
        public string FullName { get; set; } = string.Empty;

        public DateTime DateOfBirth { get; set; }

        [Required]
        [MaxLength(10)]
        public string Gender { get; set; } = string.Empty;

        [Phone]
        [MaxLength(20)]
        public string PhoneNumber { get; set; } = string.Empty;

        [MaxLength(250)]
        public string Address { get; set; } = string.Empty;

        public string? MedicalHistory { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public Guid? UserId { get; set; }

        // Navigation
        public User? User { get; set; }
        public ICollection<Appointment> Appointments { get; set; } = new List<Appointment>();
        public ICollection<MedicalRecord> MedicalRecords { get; set; } = new List<MedicalRecord>();
        public ICollection<Payment> Payments { get; set; } = new List<Payment>();
    }
}

File: src/frontend/src/types/patient.ts
export interface Patient {
  id: string;
  userId: string;
  fullName: string;
  dateOfBirth: string;
  gender: string;
  age?: number;
  phoneNumber: string;
  userEmail?: string;
  address?: string;
  medicalHistory?: string;
  avatarUrl?: string;
  createdAt: string;
  updatedAt: string;
  isActive: boolean;
  isEmailVerified: boolean;
  user?: {
    id: string;
    email: string;
    isActive: boolean;
    isEmailVerified: boolean;
  };
}

export interface UpdatePatientDTO {
  fullName?: string;
  dateOfBirth?: string;
  gender?: string;
  phoneNumber?: string;
  address?: string;
  medicalHistory?: string;
}